packages/main/src/Calendar.js

```diff
@@ -416,11 +490,83 @@ class Calendar extends UI5Element {
 				this._hideMonthPicker();
 			}
 		}
+
+		if (isTabNext(event)) {
+			this._handleTabNext(event);
+		}
+
+		if (isTabPrevious(event)) {
+			this._handleTabPrevous(event);
+		}
+	}
+
+	_handleTabNext(event) {
+		const target = event.target;
+
+		if (target.tagName === "UI5-DAYPICKER" || target.tagName === "UI5-MONTHPICKER" || target.tagName === "UI5-YEARPICKER") {
+			if (this.monthButton.getAttribute("hidden") === null) {
+				this.monthButton.focus();
+			} else {
+				this.yearButton.focus();
+			}
+			event.preventDefault();
+		} else if (target.tagName === "UI5-CALENDAR-HEADER" && event.path[0].getAttribute("data-sap-show-picker") === "Month") {
+			this.yearButton.focus();
+			event.preventDefault();
+		} else {
+			this._setPickerCurrentTabindex(-1);
+		}
+	}
+
+	_handleTabPrevous(event) {
+		const target = event.target;
+
+		if (target.tagName === "UI5-CALENDAR-HEADER" && event.path[0].getAttribute("data-sap-show-picker") === "Month") {
+			this._moveFocusToPickerContent();
+			event.preventDefault();
+		} else if (target.tagName === "UI5-CALENDAR-HEADER" && event.path[0].getAttribute("data-sap-show-picker") === "Year") {
+			if (this.monthButton.getAttribute("hidden") === null) {
+				this.monthButton.focus();
+			} else {
+				this._moveFocusToPickerContent();
+			}
+			event.preventDefault();
+		}
+	}
+
+	_moveFocusToPickerContent() {
+		if (!this._oMonth._hidden) {
+			this.dayPicker._itemNav.focusCurrent();
+		} else if (!this._monthPicker._hidden) {
+			this.monthPicker._itemNav.focusCurrent();
+		} else {
+			this.yearPicker._itemNav.focusCurrent();
+		}
+	}
+
+	_onfocusout(event) {
+		this.monthButton.setAttribute("tabindex", "-1");
```

Comment:
Could  you change the tabindex through the state and the template.
For example in the CalendarHeader.hbs the month and the year button has tabindex="0" hardcoded. But, you can use a variable, private property, that might changee because of this or that, and as a result re-renders the calendar header and we have "0" or "-1".

Reply:
Done. Currently there is no need to provide different tab index for the month and year picker buttons in the calendar header component.